static/js/app.js

```diff
@@ -1,3 +1,49 @@
-// $('#input-forward').on('click', function(){
-//   $(this).load("{{ request.route_url('edit-entry', id=entry.id) }}")
-// })
+// Toggle Add New element from any page.
+$('#ajax_new').on('click', function(e) {
+  e.preventDefault();
+  $('#a_new_entry').slideToggle();
+})
+
+// Request ajax call to DB for Edit Entry from Detail page.
+$('#input-forward').on('click', function(e) {
+  e.preventDefault();
+  $('#a_new_entry').slideToggle();
```

Comment:
I don't think you want `#a_new_entry` here 
```html
<section id='a_new_entry' hidden>
      <form action="{{ request.route_url('add') }}" method="POST" class="add_entry">
...
```
Your edit function is adding a new entry, not modifying an old one.